### DISCUSSION TITLE
Header for Login??

### DISCUSSION THREAD

**Discussion by JohnLar42** [ipxe/ipxe#629] [Ideas]
It would be nice if we could specify a header and some text inside login
Currently there is no way of customizing the text in the login prompt aside from presetting the username (meh)

**Comment by stappersg** [comment]
On Mon, Apr 04, 2022 at 06:06:10AM -0700, JohnDavidLarsen wrote:
> It would be nice  ...  (meh)


Acknowlegde on the "Please add Header for Login" wish.

And I admit that I don't understand the request.
Probably because I don't have the problem that it should solve.

My advice is to make an elaborate version of the wish
now that is confirmed there are people who do see
these iPXE support requests.


Groeten
Geert Stappers
iPXE fanboy, can't be bothered that this "Github discussion"
should have been a "Github issue".
-- 
Silence is hard to parse

**Reply by JohnLar42** [reply]
Just wondering if we could customize the login page.
Currently it just says Username:  and Password:
It would be nice to have a header that says  "Welcome: Please Login with your XYZ account".
A footer might be nice for some users as well.

**Comment by stappersg** [comment]
On Mon, Apr 04, 2022 at 09:16:43AM -0700, JohnDavidLarsen wrote:
>  ... if we could customize the login page.

And where has iPXE a login page?


Yes, I would like to be smacked with an URL to "the iPXE login page".

**Reply by JohnLar42** [reply]
https://ipxe.org/cmd/login
Looks like
https://ipxe.org/_detail/screenshots/login.png

**Comment by stappersg** [comment]
So this about the `login` command for getting values for variable `username` and `password` so that they can be used further on in the netboot process.

Now that I understand the _"problem"_ better,  I add
```text
stappers@paddy:~/src/ipxe
$ find . -name '*login*[ch]'
./src/hci/tui/login_ui.c
./src/hci/commands/login_cmd.c
./src/include/ipxe/login_ui.h
stappers@paddy:~/src/ipxe
$ 
```

If I would be _stakeholder_, I would give the `login` command an optional parameter. The optional parameter would be a string.  Think `echo`  and `echo string to be printed`.

**Comment by christianbovey** [comment]
Here is a patch I wrote a few years ago that does exactly that if I understand correctly.
The "login" command can then be called with a string as a parameter.

[add_title_option_to_login_command.zip](https://github.com/ipxe/ipxe/files/8451855/add_title_option_to_login_command.zip)

**Reply by stappersg** [reply]
```text
patching file src/hci/commands/login_cmd.c
Hunk #1 succeeded at 23 (offset 4 lines).
Hunk #2 succeeded at 44 (offset 4 lines).
Hunk #3 succeeded at 56 (offset 4 lines).
patching file src/hci/tui/login_ui.c
Hunk #1 succeeded at 38 (offset 4 lines).
Hunk #2 succeeded at 56 (offset 4 lines).
Hunk #3 succeeded at 75 (offset 4 lines).
Hunk #4 succeeded at 86 (offset 4 lines).
patching file src/include/ipxe/login_ui.h
```
which became #631 

@JohnDavidLarsen please tell _the iPXE community_ how much that improves your situation.

**Reply by stappersg** [reply]
Let's see what @JohnDavidLarsen will do with this reminder.

**Reply by JohnLar42** [reply]
Thanks for the reminder.  Tested it, thanks. That definately gets me started in the right direction. I might see if I can add background image support to login screen. Or would that simply be solved by changing the Console? :/ ??

**Reply by mcb30** [reply]
Yes, you can use the existing support for background images: see https://ipxe.org/cmd/console

**Reply by JohnLar42** [reply]
Thanks, might use that instead. Need multiple lines, I see that each line is a xU. I could work from the source, but not sure of a good way to pass in multiple large arguments, so an image of text might be better.

**Comment by JohnLar42** [comment]
Thanks for the reminder, testing....

**Comment by NiKiZe** [comment]
Some recent developments announced today, see https://ipxe.org/cmd/present
